pages/Product/components/AddToCartBar/index.jsx

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
+ *
+ * This source code is licensed under the Apache 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import AddToCartButton from './components/AddToCartButton';
+import AddMoreButton from './components/AddMoreButton';
+import CartItemsCount from './components/CartItemsCount';
+import connect from './connector';
+import styles from './style';
+
+/**
+ * The AddToCartBar component.
+ */
+class AddToCartBar extends Component {
+  static propTypes = {
+    cartProductCount: PropTypes.number.isRequired,
+    handleAddToCart: PropTypes.func.isRequired,
+  };
+
+  /**
+   * Constructor.
+   */
+  constructor() {
+    super();
+
+    this.state = {
+      itemCount: 0,
+    };
+  }
+
+  /**
+   * Resets the item count if the cart is emptied.
+   * @param {Object} nextProps The next props.
+   */
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.cartProductCount === 0) {
```

Comment:
> If the user leaves the product detail page and comes back, or if he **switches to a different product variant**, the bar is reset to show the "add to cart" button again

Additionally there might be some other situations where the bar needs a reset. For example when the user has put a product into the cart, goes to the cart and reduces the amount, and goes back to the product. Then the counter differs from the count within the cart.

Also the counter does not decrease when the "addToCart" request fails.

Reply:
The counter does still increase if an `addProductsToCart` request fails. I've had this at a product with limited stock quantity. The user sees a popup with an information that the product wasn't added to the cart, but the counter increased.